skins/cat17/address-form/src/__tests__/components/AddressForm.spec.ts

```diff
@@ -1,71 +1,321 @@
+	});
+
+	it('dispatches validateInput action with the field\'s value on blur from an input field', () => {
+		const wrapper = mount(AddressForm, {
+			store,
+			localVue,
+			propsData: newTestProperties({ isCompany: false })
+		});
+		let input = wrapper.find('#first-name');
+		input.setValue('test');
+		input.trigger('blur');
+		expect(actions.validateInput).toHaveBeenCalledWith(
+			expect.anything(),
+			{
+				name: 'firstName',
+				value: 'test',
+				pattern: '^.+$',
+				optionalField: false
+			},
+			undefined // empty "Root State"
+		);
+	});
+
+	it('updates fieldErrors correctly according to the validation result', () => {
+		const wrapper = mount(AddressForm, {
+			store,
+			localVue,
+			propsData: newTestProperties({ isCompany: false })
+		});
+
+		let input = wrapper.find('#post-code');
+		input.setValue('test');
+		input.trigger('blur');
+		let comp = wrapper.vm.$options.computed;
+		if (comp !== undefined) {
+			if (typeof comp.fieldErrors === 'function') {
+				let errors = comp.fieldErrors();
+				expect(errors.postcode).toBe(false);
+			}
+		}
+	});
+
+	it('displays errors for the correct fields', () => {
+		const wrapper = mount(AddressForm, {
+			store,
+			localVue,
+			propsData: newTestProperties({ isCompany: false })
+		});
+		let title = wrapper.find('#title');
+		title.setValue('');
+		let firstName = wrapper.find('#first-name');
+		firstName.setValue('Testina');
+		let lastName = wrapper.find('#last-name');
+		lastName.setValue('von Testinson');
+		let street = wrapper.find('#street');
+		street.setValue('Testenhofen Ufer 23-24');
+		let postCode = wrapper.find('#post-code');
+		postCode.setValue('1234');
+		let city = wrapper.find('#city');
+		city.setValue('');
+		let country = wrapper.find('#country');
+		country.setValue('DE');
+		let comp = wrapper.vm.$options.computed;
+		if (comp !== undefined) {
+			if (typeof comp.fieldErrors === 'function') {
+				let errors = comp.fieldErrors();
+				expect(errors.title).toBe(true);
+				expect(errors.firstName).toBe(false);
+				expect(errors.lastName).toBe(false);
+				expect(errors.street).toBe(false);
+				expect(errors.postcode).toBe(false);
+				expect(errors.city).toBe(true);
+				expect(errors.country).toBe(false);
+			}
+		}
+	});
+
+	it('saves user input to the respective part of formData', () => {
+		const wrapper = mount(AddressForm, {
+			store,
+			localVue,
+			propsData: newTestProperties({ isCompany: false })
+		});
+		let title = wrapper.find('#title');
+		title.setValue('Prof. Dr.');
+		let firstName = wrapper.find('#first-name');
+		firstName.setValue('Testina');
+		let lastName = wrapper.find('#last-name');
+		lastName.setValue('von Testinson');
+
+		expect(wrapper.vm.$data.formData.title.value).toBe('Prof. Dr.');
+		expect(wrapper.vm.$data.formData.firstName.value).toBe('Testina');
+		expect(wrapper.vm.$data.formData.lastName.value).toBe('von Testinson');
+	});
+
+	it('sets hidden field addressType "firma" for company form', () => {
```

Comment:
I'm sorry that we have to perpetuate the German identifiers here, for the sake of the server-side code. I've created a ticket to translate the identifiers, but it's a major effort for very little gain, so I fear this improvement might never happen: https://phabricator.wikimedia.org/T221044